tensorflow/python/ops/nn_grad.py

```diff
@@ -203,6 +203,48 @@ def _BiasAddGrad(op, received_grad):
   return (received_grad, gen_nn_ops.bias_add_grad(out_backprop=received_grad,
                                                   data_format=data_format))
 
+@ops.RegisterGradient("BiasAddGrad")
+def _BiasAddGradGrad(op, received_grad):
+  """Gradient for the BiasAddGrad op.
+
+  Args:
+    op: BiasAddGrad op for which we are calculating gradients.
+    received_grad: The gradients passed to the BiasAddGrad op.
+    
+  Returns:
+    A single gradient Tensor for the input to BiasAddGrad (which
+    is the gradient of the bias term in BiasAdd)
+  """
+  
+  try:
+    data_format = op.get_attr("data_format")
+  except ValueError:
+    data_format = None
+    
+  #zeros = array_ops.zeros_like(op.inputs[0])
+  #return gen_nn_ops._bias_add(zeros, received_grad, data_format=data_format)
+  
+  shape = array_ops.shape(op.inputs[0])
+  rank = array_ops.rank(op.inputs[0])
+  bias_shape = array_ops.shape(received_grad)
```

Comment:
Some crude benchmarks:
[benchmark.zip](https://github.com/tensorflow/tensorflow/files/479926/benchmark.zip)

The benchmarks were done on a workstation with an i7-965 (yes, we laughed too when we got it) and a GTX 1080.

The results are a bit odd. Using `bias_add` with `zeros_like` seems to give a negligible to moderate performance advantage in most cases, but it is 2-5x slower on the CPU with statically determined shapes (it looks like something is not being optimized properly here). I did notice that on the GPU, using `bias_add` seemed to result in somewhat higher GPU usage overall.

I'm inclined to stay with `tf.tile` for now, since (1) it is a more direct solution (using `bias_add` seems extremely hacky), (2) its performance is more consistent, and (3) you mentioned that the performance of `tf.tile` might be improved in the future.

Of course, the optimal solution would be to write another kernel for this operation (and then we could finally end the chain of gradient ops, since this operation's gradient is identical to `bias_add_grad`), but I'm not sure if that is worthwhile for something that most machine learning people aren't going to need.